### ISSUE TITLE
[java] UnusedImports false positive with static ondemand import with method call

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1404]
**Affects PMD Version:** 6.9.0-SNAPSHOT

**Rule:** UnusedImports

**Description:** PMD flags the static import as unused, when it isn't.
This seems to be something different than #1209 . We obviously don't see, that the constant is used in a method call.

**Code Sample demonstrating the issue:**

```java
package net.sourceforge.pmd.lang.java.rule.bestpractices.unusedimports;

import static net.sourceforge.pmd.lang.java.rule.bestpractices.unusedimports.ClassWithStringConstants.*;

public class ClassWithImport {

    public static void main(String[] args) {
        if (CONST1.equals("a")) {
            System.out.println("CONST1 is a");
        }
        // using this works fine:
        // System.out.println("CONST1 is " + CONST1);
        // also this works:
        // if ("a".equals(CONST1)) { ... }
    }
}
```

With ClassWithStringConstants being:

```java
package net.sourceforge.pmd.lang.java.rule.bestpractices.unusedimports;

public class ClassWithStringConstants {

    private ClassWithStringConstants() {
        // Utility class
    }

    public static final String CONST1 = "a";
    public static final String CONST2 = "b";
}
```

**Running PMD through:** any

**Comment by oowekyala** [comment]
@adangel Thanks for fixing this :)
